homeassistant/helpers/entity_values.py

```diff
@@ -0,0 +1,42 @@
+"""A class to hold entity values."""
+import fnmatch
+import re
+
+from homeassistant.core import split_entity_id
+
+
+class EntityValues(object):
+    """Class to store entity id based values."""
+
+    def __init__(self, exact=None, domain=None, glob=None):
+        """Initialize an EntityConfigDict."""
+        self._cache = {}
+        self._exact = exact
+        self._domain = domain
+
+        if glob is not None:
+            glob = {re.compile(fnmatch.translate(key)): value
+                    for key, value in glob.items()}
+
+        self._glob = glob
+
+    def get(self, entity_id):
```

Comment:
1) 
I think long term we should either support deep stuff properly or forbid it.
Even if core doesn't use it - it can be used by custom components, platforms and UI.

Meanwhile I'm ok with this being as-is until this whole thing stabilizes more.

2)
More specific should override the more general.
Glob can be either more specific than domain: `light.bedroom*` or more general than doman: `*sensor` or both: `*sensor.bedroom*`
I think the first usecase is the most useful, thus I'm for "domain, glob, id" .

That said, if the glob order is preserved, whoever wants to make write a least-strong-domain-rule can write glob `light.*`.
In that case "glob, domain, id" would indeed make more sense.